.github/workflows/dependencies/updater.py

```diff
@@ -14,437 +19,551 @@
 # Dry run flag
 DRY_RUN = os.environ.get("DRY_RUN", "0") == "1"
 
-import timeit
+# utils for tag comparison
+BASEVERSION = re.compile(
+    r"""[vV]?
+    (?P<major>0|[1-9]\d*)
+    (\.
+    (?P<minor>0|[1-9]\d*)
+    (\.
+    (?P<patch>0|[1-9]\d*)
+    )?
+    )?
+    """,
+    re.VERBOSE,
+)
```

Comment:
This regex here fails when the version numbers begin with `0`, in which case the result are 0s:
```py
>>> coerce("v2.03.4")
Version(major=2, minor=0, patch=0, prerelease=None, build=None)
```

The `0|[1-9]` part should go inside an additional parenthesis:
```suggestion
BASEVERSION = re.compile(
    r"""[vV]?
    (?P<major>(0|[1-9])\d*)
    (\.
    (?P<minor>(0|[1-9])\d*)
    (\.
    (?P<patch>(0|[1-9])\d*)
    )?
    )?
    """,
    re.VERBOSE,
)
```

With this change the `coerce` function works again:
```py
>>> coerce("v2.3.4")
Version(major=2, minor=3, patch=4, prerelease=None, build=None)
>>> coerce("v2.03.4")
Version(major=2, minor=3, patch=4, prerelease=None, build=None)
>>> coerce("zsh2.03.4")
Version(major=2, minor=3, patch=4, prerelease=None, build=None)
>>> coerce("zsh-2.03.4")
Version(major=2, minor=3, patch=4, prerelease=None, build=None)
```

I see that the `semver` package [has a parse function](https://pypi.org/project/semver/), but I have tested it and I understand you're doing it this way because it does not work well with prefixed strings on the versions, so we'd have to remove that from tags either way. Am I right?
```py
>>> Version.parse("v2.3.4")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/.../ohmyzsh/.github/workflows/dependencies/venv/lib/python3.12/site-packages/semver/version.py", line 646, in parse
    raise ValueError(f"{version} is not valid SemVer string")
ValueError: v2.3.4 is not valid SemVer string
```

Reply:
It does not work with prefixed strings, and it doesn't work either with `2.0`, which lacks a `patch` section. Theoretically, in `semver`, `M.m.p` are compulsory.

Reply:
I just took that `coerce` function from [`python-semver` docs](https://python-semver.readthedocs.io/en/latest/advanced/deal-with-invalid-versions.html).